qiskit_experiments/database_service/json.py

```diff
@@ -109,27 +308,67 @@ class ExperimentEncoder(json.JSONEncoder):
     """JSON Encoder for Numpy arrays and complex numbers."""
 
     def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
-        if isinstance(obj, (np.ndarray, complex)):
-            return serialize_safe_float(obj)
+        if isinstance(obj, complex):
+            return _serialize_safe_float(obj)
+        if isinstance(obj, set):
+            return {"__type__": "set", "__value__": list(obj)}
+        if isinstance(obj, np.ndarray):
+            value = _serialize_and_encode(obj, np.save, allow_pickle=False)
```

Comment:
Not sure if it applies here, but runtime also uses `np.save`, and someone complained the result is not readable on IQX...

Reply:
We do special handling for the UI display fields of results (which don't support Arrays at the moment anyway, just floats), so I think this should be ok.